src/main/PlagueMain.java

```diff
@@ -1084,38 +1128,29 @@ void loadMap(String[] args) {
         }
 
         logic.reset();
-        mindustry.maps.Map map = maps.customMaps().get(currMap);
         Log.info("Loading map " + map.name());
 
         world.loadMap(map);
 
-        Tile tile = state.teams.cores(Team.malis)
-                .find(build -> build.block == Blocks.coreNucleus || build.block == Blocks.coreFoundation
-                        || build.block == Blocks.coreAcropolis).tile;
-
         // Make cores and power source indestructible
-        state.teams.cores(Team.malis).each(coreBuild -> coreBuild.health = Integer.MAX_VALUE);
-        for (int x = 0; x < world.width(); x++) {
-            for (int y = 0; y < world.height(); y++) {
-                Tile t = world.tile(x, y);
-                if (t.build != null && t.build.block.equals(Blocks.powerSource)) {
-                    t.build.health = Integer.MAX_VALUE;
-                }
-            }
-        }
-
-        isSerpulo = tile.build.block != Blocks.coreAcropolis;
-
-        plagueCore[0] = tile.x;
-        plagueCore[1] = tile.y;
+        Team.malis.cores().each(coreBuild -> {
+            coreBuild.health = Float.MAX_VALUE;
+            coreBuild.items.clear();
+        });
+        isSerpulo = PlagueData.serpuloCores.contains(Team.malis.cores().get(0).block);
+        world.tiles.forEach(t -> {
+            if (t.build != null && t.build.block.equals(Blocks.powerSource))
+                t.build.health = Float.MAX_VALUE;
+        });
+        plagueCore = new Vec2(map.width / 2, map.height / 2); // center
         world.beginMapLoad();
         PlagueGenerator.defaultOres(world.tiles, isSerpulo);
 
         world.endMapLoad();
-
-        rules.enemyCoreBuildRadius = 75 * 7;
-        rules.hiddenBuildItems = map.rules().hiddenBuildItems;
+        rules.hiddenBuildItems = (isSerpulo ? Items.erekirOnlyItems : PlagueData.serpuloOnlyItems).asSet();
+        rules.bannedBlocks = isSerpulo ? ObjectSet.with() : ObjectSet.with(Blocks.plastaniumConveyor);
         rules.hideBannedBlocks = true;
+
```

Comment:
Since this isn't part of the diffs, I'll allow it despite having no knowledge of the firstRun boolean